pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4455,6 +4506,83 @@ int BPF_KPROBE(trace_security_inode_mknod)
     return events_perf_submit(&data, SECURITY_INODE_MKNOD, 0);
 }
 
+SEC("kprobe/do_splice")
+TRACE_ENT_FUNC(do_splice, DIRTY_PIPE_SPLICE);
+
+SEC("kretprobe/do_splice")
+int BPF_KPROBE(trace_ret_do_splice)
+{
+    args_t saved_args;
+    if (load_args(&saved_args, DIRTY_PIPE_SPLICE) != 0) {
+        // missed entry or not traced
+        return 0;
+    }
+
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    // dirty_pipe_splice is a splice to a pipe which results that the last page copied could be modified (the
+    // PIPE_BUF_CAN_MERGE flag is on in the pipe_buffer struct).
+    if (!event_chosen(DIRTY_PIPE_SPLICE))
+        return 0;
+
+    struct file *out_file = (struct file*)saved_args.args[2];
+    struct inode *out_inode = READ_KERN(out_file->f_inode);
+    struct pipe_inode_info *out_pipe = READ_KERN(out_inode->i_pipe);
+    // Check that output is a pipe
+    if (!out_pipe) {
```

Comment:
You are very right.
It crossed my mind when writing it but I forgot to test if it might prove to be problematic. There is a more hermetic way to do this check, but I was a bit lazy to implement it (it requires to use kallsyms which wasn't available until now).
Anyways, I will implement the other way (as implemented by the [get_pipe_info](https://elixir.bootlin.com/linux/latest/source/fs/pipe.c#L1367) function).